modules/apps/27-interchain-accounts/host/keeper/keeper_test.go

```diff
@@ -47,21 +56,32 @@ func (suite *KeeperTestSuite) SetupTest() {
 	suite.chainC = suite.coordinator.GetChain(ibctesting.GetChainID(3))
 }
 
-func NewICAPath(chainA, chainB *ibctesting.TestChain) *ibctesting.Path {
+func NewICAPath(chainA, chainB *ibctesting.TestChain, encoding string) *ibctesting.Path {
```

Comment:
I didn't because the original proto didn't include an enum. I'll implement it if others agree

Reply:
I think it would be nice to make this change, previously I don't think the encoding type as a string was exposed to any public interfaces, but these changes will make that the case. Also happy with a follow up PR making only these changes.

Reply:
+1 for the enum type!

Reply:
I've looked into using enums and will probably make this change soon but here is something I've noticed:

- You cannot customise the stringer by using gogoproto extensions. But we need a custom stringer for backwards compatibility (this is because we need the `ENCODING_PROTO3` enum variant to become `"proto3"`). Which means we must disable the default stringer (which is something the extensions can do), and write our own stringer in `metadata.go`.

This feels over-complicated for this functionality. Nevertheless, I'll make the change and see what happens.

Reply:
Actually sorry, this works: (but it looks weird due to using non-capitalized enum variants):
```protobuf
enum Encoding {
  option (gogoproto.goproto_enum_prefix) = false;

  // PROTOBUF defines the protobuf encoding
  proto3 = 0 [(gogoproto.enumvalue_customname) = "ENCODING_PROTO3"];
  // Proto3JSON defines the proto3 JSON encoding
  proto3json = 1 [(gogoproto.enumvalue_customname) = "ENCODING_PROTO3JSON"];
}
```

Reply:
Finally, after attempting to implement this, I realized this would require reopening of the merged PR to the main, see [#3967](https://github.com/cosmos/ibc-go/pull/3967). Which leads me to think that it is best done in another PR to main, either after this is merged, or before. So I will not push the changes here. I will discuss this briefly in the engineering call

Reply:
Agree it should be discussed in a separate pr. As a side note, enums are encoded differently between standard json and proto3 json. Currently ICA uses proto3 json for encoding the metadata, if it used standard json this change wouldn't be backwards compatible

The enum string value for proto3 must be the same as the existing string otherwise we will break cross compatibility for opening channels

I would also not backport this change